### PR TITLE
Add a dependsOn condition to wait for successful completion

### LINKED ISSUE #11: investigate support for init container
**What is the problem you're trying to solve**
init container(s) run before service container in sequence and can prepare environment. Think "containerized entrypoint"

We might want to support such init containers in compose spec, with obvious mapping for Kubernetes

**Describe the solution you'd like**
syntax proposal and lifecycle document

**Additional context**
See https://github.com/docker/compose/issues/6855

### PR DESCRIPTION
**Pull request by mikesir87**
**What this PR does / why we need it**:

Updates the spec to add a `dependsOn.condition` that waits for the successful completion of a dependent service. This is important to help support init container workflows.

**Which issue(s) this PR fixes**:

Fixes #11

### CODE REVIEW

**Comment by ndeloof** [comment]
LGTM, two notes:
* I'm not native english but https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdependency.html uses "COMPLETE" and not "FINISHED" for such a state, maybe the former would be a better noun? 
*  if we adopt this I'd expect we also define a `service_completed` value that would not check exit code

**Comment by mikesir87** [comment]
> I'm not native english but https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdependency.html uses "COMPLETE" and not "FINISHED" for such a state, maybe the former would be a better noun?

I had the same debate myself and like `completed` more. So, I switched it to `service_completed_successfully`

> if we adopt this I'd expect we also define a service_completed value that would not check exit code

Honestly, I'm trying to think of the scenario where someone might need that. If the dependent service failed, when would you still want the dependent service to run? Seems quite risky to me, but probably only because I can't think of a use case for it.

**Review by ndeloof — APPROVED** [review]
don't forget to update the json schema to reflect this new value

**Comment by mikesir87** [comment]
> don't forget to update the json schema to reflect this new value

And done. Thanks @ndeloof!

File: spec.md [hunk 1]
```diff
@@ -598,6 +598,8 @@ expressed in the short form.
   - `service_healthy`: specifies that a dependency is expected to be "healthy"
     (as indicated by [healthcheck](#healthcheck)) before starting a dependent
     service.
+  - `service_completed_successfully`: specifies that a dependency is expected to run 
+    to successful completion before starting a dependent service.
```
Comment (EricHripko) [inline spec.md:602]: Apologies if this is overly pedantic, but could we specify that `successful completion` means exiting with code `0`?

**Comment by EricHripko** [comment]
also CC @kohidave

File: spec.md (hunk 1 above)
Comment (ndeloof) [reply spec.md:602]: we discussed this point during compose-spec meeting and prefered **not** to get into this Linux process implementation detail. Compose model could be applied to distinct things (let's say, VMs, unikernels or WASM payload) which might have a distinct way to notify the platform execution has been successful.

Comment (EricHripko) [reply spec.md:602]: Fair point, resolving 🙂

**Review by EricHripko — APPROVED** [review]
Looks good ✅

**Pull request merged**